### ISSUE TITLE
Adminstrators should have a way of removing VLANs from the pool

### ISSUE THREAD

**Issue by zenhack** [CCI-MOC/haas#446]
Once the database is initialized, there's no way to get VLANs *out* of the database. Even changing the config file won't have an effect.

**Comment by gsilvis** [comment]
I've argued before that VLANs should be added to and removed from the pool with API calls rather than the config file.  But the behavior here is fairly non-obvious.  What happens if I delete a VLAN that has a network allocated to it, for instance?

**Comment by zenhack** [comment]
The right behaviour in that case is probably to just refuse to delete it without first deleting the network.

Also, since vlans are a driver-specific notion, we'd have to either (a) figure out how to make the api calls generic, or (b) decide how extension-supplied api calls should work.

I can't think of a better solution though.

**Comment by gsilvis** [comment]
Yeah, refusing to delete is in line with most other HaaS commands.

**Comment by naved001** [comment]
> (b) decide how extension-supplied api calls should work.

We use database authentication and it's APIs quite frequently and when it's not loaded it just returns a 404. Why can't we do the same for other extension supplied api calls?

**Comment by zenhack** [comment]
The thing that bugged me is that we have no namespacing mechanism for api calls, so really extensions shouldn't be defining them until we have a clearer notion of how to make sure they don't trample on eachother.

In theory, one might have extensions that are developed and maintained separately -- imagine an organization that has their own internal auth(z) system they want to integrate with, that doesn't really make sense to upstream into HIL core. Same thing with switches that we don't have the ability to test against; I'd rather not maintain code I can't test, but right not it's perfectly possible for somone to write a separate python package with another driver. This is why the extensions are specified as full module paths.

`docs/extensions.rst` has a description of the mechansim, including an overview of what extensions are allowed to touch. But we've never really referenced or checked it very carefully when doing stuff in extensions, and everything is basically subject to change at any time, so I don't know how practical all of this really is.

**Comment by naved001** [comment]
> so really extensions shouldn't be defining them until we have a clearer notion of how to make sure they don't trample on eachother.

That makes a lot of sense, which is why you were careful in adding APIs for db auth backend.
